btrack/napari/main.py

```diff
@@ -80,6 +80,14 @@ def create_btrack_widget() -> btrack.napari.widgets.BtrackWidget:
         lambda selected: select_config(btrack_widget, all_configs, selected),
     )
 
+    # Disable the Optimiser tab if unchecked
+    for tab in range(btrack_widget._tabs.count()):
+        if btrack_widget._tabs.tabText(tab) == "Optimiser":
+            break
+    btrack_widget.enable_optimisation.toggled.connect(
+        lambda is_checked: btrack_widget._tabs.setTabEnabled(tab, is_checked)
+    )
```

Comment:
This was the best way I could get it without relying on index @p-j-smith

Reply:
@paddyroddy I think that's an okay way to do it. Or if you prefer you can get the index of one of the tabs using the [widget we put in that tab](https://github.com/quantumjot/btrack/blob/e2ee8d82713496655be915d7659dcde82a3de090/btrack/napari/widgets/create_ui.py#L137-L139). As we're not storing the widget itself, you could use one of the child widgets (e.g. [`hypotheses`](https://github.com/quantumjot/btrack/blob/e2ee8d82713496655be915d7659dcde82a3de090/btrack/napari/widgets/_optimiser.py#L34)) and find its parent:

```python
optimiser_tab_index = btrack_widget._tabs.indexOf(btrack_widget.hypotheses.parentWidget())
```

Reply:
Yeah, I saw that, but I'm not convinced it's any better! Especially as some widgets have moved tabs